xbmc/pvr/PVRDatabase.cpp

```diff
@@ -229,6 +229,12 @@ void CPVRDatabase::UpdateTables(int iVersion)
 
   if (iVersion < 37)
     m_pDS->exec("ALTER TABLE channelgroups ADD iLastOpened integer");
+
+  bool bisMySQL = StringUtils::EqualsNoCase(
+      CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_databaseTV.type, "mysql");
+
+  if (iVersion < 38 && bisMySQL)
+    m_pDS->exec("ALTER TABLE channelgroups MODIFY COLUMN iLastOpened BIGINT UNSIGNED");
```

Comment:
By taking this approach you are going to have some schema divergence going forwards. What I mean is this:
- users on SQLite that have migrated from v37 or before will have iLastOpened INTEGER
- users on SQLite with a fresh install of v38, and all those on MySQL will have iLastOpened BIGINT UNSIGNED

Maybe this doesn't matter it will make no difference to how Kodi functions, and the migration/fresh routes can often result in different column orders. But it could be better from a human and support perspective to have a consistent type definitions for columns across all users. 

The alternative to using COLUMN MODIFY for MySQL only is to create a table channelgroups_new with iLastOpened BIGINT UNSIGNED and then INSERT INTO using column values from channelgroups. More lines of code and probably takes a litte longer to migrate. 

Totally up to @ksooo which you choose,  I'm just pointing out consequnces.